src/ast/mod.rs

```diff
@@ -2980,6 +2980,32 @@ pub enum Statement {
         show_options: ShowStatementOptions,
     },
     /// ```sql
+    /// SHOW [ TERSE ] OBJECTS [ LIKE '<pattern>' ]
+    ///                        [ IN
+    ///                              {
+    ///                                ACCOUNT                                         |
+    ///    
+    ///                                DATABASE                                        |
+    ///                                DATABASE <database_name>                        |
+    ///
+    ///                                SCHEMA                                          |
+    ///                                SCHEMA <schema_name>                            |
+    ///                                <schema_name>
+    ///
+    ///                                APPLICATION <application_name>                  |
+    ///                                APPLICATION PACKAGE <application_package_name>  |
+    ///                              }
+    ///                        ]
+    ///                        [ STARTS WITH '<name_string>' ]
+    ///                        [ LIMIT <rows> [ FROM '<name_string>' ] ]
+    /// ```
+    /// Snowflake-specific statement
+    /// <https://docs.snowflake.com/en/sql-reference/sql/show-objects>
+    ShowObjects {
+        terse: bool,
+        show_options: ShowStatementOptions,
```

Comment:
Could we use a struct to wrap the arguments? we're currently trying to move away from the anonymous struct pattern (https://github.com/apache/datafusion-sqlparser-rs/issues/1204)
Something like
```rust
struct ShowObjects {
         terse: bool,
        show_options: ShowStatementOptions,
}
Statement::ShowObjects(ShowObjects)
```